backend/packages/Upgrade/src/api/Algorithms.ts

```diff
@@ -0,0 +1,200 @@
+import seedrandom from 'seedrandom';
+import { ConditionPayloadDTO } from './DTO/ConditionPayloadDTO';
+import { Experiment } from './models/Experiment';
+import { CONDITION_ORDER, EXPERIMENT_TYPE, IExperimentAssignmentv5, AssignedCondition, IPayload } from 'upgrade_types';
+import { FactorDTO } from './DTO/FactorDTO';
+import { ExperimentCondition } from './models/ExperimentCondition';
+
+export function withInSubjectType(
+  experiment: Experiment,
+  conditionPayloads: ConditionPayloadDTO[],
+  site: string,
+  target: string,
+  factors: FactorDTO[],
+  userID: string,
+  monitoredDecisionPointLogsLength: number
+): IExperimentAssignmentv5 {
+  const assignedData = convertToAssignedCondition(experiment, conditionPayloads, site, target, factors);
+  let assignedConditionsArray = assignedData;
```

Comment:
Not required. Just convert `const assignedData`to `let assignedData`

Reply:
Why not make this into `Switch` statement? It will be more readable

Reply:
code updated